lib/base.js

```diff
@@ -54,6 +59,72 @@ Base.prototype._checkStatusForError = function(statusCode, body) {
     }
 };
 
+Base.prototype.register = function({username, password, fields}, done) {
+    if (!username) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'username' required for Base#register", 404);
+    }
+    if (!password) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'password' required for Base#register", 404);
+    }
+    if (this._airtable._endpointUrl.includes('https://api.airtable.com')) {
+        throw new AirtableError('NOT_AUTHORIZED', 'Base#register cannot be used with api.airtable.com. Please configure this Airlock client to use an Airlock endpoint URL.', 403);
+    }
+    if (this._user !== null || this._getLocalStorage('user') !== null) {
+        throw new AirtableError('NOT_AUTHORIZED', 'Already logged in');
+    }
+    runAction(this, 'post', '/__DANGEROUSLY__USE__TABLE__TO__LET__USERS__REGISTER', {}, {username, password, fields}, (err, data) => {
+        this._user = {
+            username: data.body.username,
+            token: data.body.token
+        };
+        this._setLocalStorage('user', this._user);
+        done(err, data);
+    });
+};
+
+Base.prototype.login = function({username, password}, done) {
+    if (!username) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'username' required for Base#login", 404);
+    }
+    if (!password) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'password' required for Base#login", 404);
+    }
+    if (this._airtable._endpointUrl.includes('https://api.airtable.com')) {
+        throw new AirtableError('NOT_AUTHORIZED', 'Base#login cannot be used with api.airtable.com. Please configure this Airlock client to use an Airlock endpoint URL.', 403);
+    }
+    if (this._user !== null) {
+        throw new AirtableError('NOT_AUTHORIZED', 'Already logged in');
+    }
+    runAction(this, 'post', '/__DANGEROUSLY__USE__TABLE__TO__LET__USERS__LOGIN', {}, {username, password}, (err, data) => {
+        this._user = {
+            username: data.body.username,
+            token: data.body.token
+        };
+        this._setLocalStorage('user', this._user);
+
+        done(err, data);
```

Comment:
Same here! See https://www.notion.so/calblueprint/User-Authentication-and-Token-Management-72067d56addf4959a152bfbc53bca089